doc/source/ray-core/monitoring-debugging/troubleshoot-failures.rst

```diff
@@ -0,0 +1,73 @@
+Troubleshooting Failures
```

Comment:
All contents are just moved from original `troubleshooting.rst`. I will improve the contents in the follow-up PR